mobile-app/app/screens/WalletNavigator/screens/CreateWallet/OnboardingNetworkSelectScreen.tsx

```diff
@@ -4,10 +4,10 @@ import { RowNetworkItem } from '@components/RowNetworkItem'
 import { translate } from '@translations'
 import * as React from 'react'
 import { View } from 'react-native'
-import * as Updates from 'expo-updates'
+import { getReleaseChannel } from '@api/releaseChannel'
```

Comment:
We should separate out the release channel changes first

Reply:
I added it because release channel is only updated on eas builds (so all electron builds are considered as dev builds).

Reply:
@fuxingloh ok I have separated the release channel commit - https://github.com/DeFiCh/wallet/pull/1224

Reply:
could the PR be updated to target the main first, I am thinking about whether it can be refactored into a Context for inversion of control. Provided once and injected everywhere, now we are pulling it causing complexity in dependency injecting

Reply:
Oh damn I have my commits there haha wait let me create another one

Reply:
@fuxingloh updated - https://github.com/DeFiCh/wallet/pull/1229/files

Reply:
Hmmm we can do that also, something like Environment Context

Reply:
Let me check

Reply:
I am also very curious about the thoughts of using Context via the `Provider` and `Consumer` patterns in this project. It's certainly some level of implicit injection to it. I always believe it helps the project in the long run where you don't need to worry managing dependency and it's outsourced to the context and provided where applicable.

Reply:
Generally agree that it should be beneficial. Some hidden "cons" may need extra awareness

- redundant workload introduced when the context is not even used more than once
- when context is available but not being used effeciently

IMO these API injections are not needed can be embedded within the provider. Provider should be able to construct them using the `useSecureStorage`
![image](https://user-images.githubusercontent.com/44192990/140476672-1a865cb5-1df4-43ec-85f9-f1f6d77d003b.png)

Reply:
@ivan-zynesis I agree, it's just that some of our functions (e.g. secured_storage, etc.)  are not yet context (therefore, cannot use hooks at the meantime). So we don't really have a choice but to pass it as props (if we want to re-use them). It will be cleaner once we have refactored most of it to be context.